### PR TITLE
Release 118.0.0

### LINKED ISSUE #3916: Bring `TokenDetectionController` up-to-date and release new version of `assets-controllers`
## Explanation

- [ ] Update `TokenDetectionController` with changes made to the extension `detectTokensController` since #1813 was closed.
  - Diff: [MetaMask/metamask-extension@`5d285f7be5f7be981995dfa725aad97d81cc990a..85cd1c89039e900b452edb704ec37e9ccbd3e76a`#diff-323d0cf464](https://github.com/MetaMask/metamask-extension/compare/5d285f7be5f7be981995dfa725aad97d81cc990a..85cd1c89039e900b452edb704ec37e9ccbd3e76a#diff-323d0cf46498be3850b971474905354ea5ccf7fa13745ad1e6eba59c5b586830)
  - https://github.com/MetaMask/metamask-extension/pull/22898
  - https://github.com/MetaMask/metamask-extension/pull/22814

  
- [ ] Release merged `TokenDetectionController`
  - https://github.com/MetaMask/core/pull/3918
- [ ] Update `@metamask/assets-controllers` versions to v26.0.0 or higher in downstream controllers.
  - [ ] `metamask-extension`
  - [ ] ~`metamask-mobile`~

Further updates to the code may be necessary to reflect the changed API of `TokenDetectionController`, and its migration into the core repo.

### PR DESCRIPTION
**Pull request by MajorLift**
## Explanation

Releases new major version of `@metamask/assets-controllers`.

- Blocked by 
  - #3923 
  - https://github.com/MetaMask/core/pull/3915
  - https://github.com/MetaMask/core/pull/3925
- Partially fixes #3916

### CODE REVIEW

**Review by mcmire — COMMENTED** [review]
Release 116 is out, so this is ready to be rebased. I've also left some suggestions below.

File: packages/assets-controllers/CHANGELOG.md
[the remark this answers is not in the document]
Comment (MajorLift) [reply packages/assets-controllers/CHANGELOG.md:1]: Thanks for pointing these out! Applied in https://github.com/MetaMask/core/pull/3918/commits/e1cffb4d4c0eaaf7f2705fcbf39e56128680c352

**Comment by MajorLift** [comment]
- Resetting to v118 release candidate to make v117 available for urgent transaction-controller release by wallet api team. 
- Also, recent updates to `DetectTokensController` need to be ported into `TokenDetectionController` before it can be released (#3923).

**Comment by MajorLift** [comment]
Closing to avoid blocking other more urgent release. Will reopen once https://github.com/MetaMask/core/pull/3923 is merged.

**Comment by MajorLift** [comment]
Superseded by #4007

**Pull request closed without merging**